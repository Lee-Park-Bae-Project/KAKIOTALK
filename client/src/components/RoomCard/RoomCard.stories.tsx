import React from 'react';
import RoomCard from 'components/RoomCard';
import faker from 'faker';
import styled from 'styled-components';

export default {
  title: 'Component/RoomCard',
  component: RoomCard,
};

export const Default = () => {
  const userList = [
    faker.helpers.userCard().name,
    faker.helpers.userCard().name,
    faker.helpers.userCard().name,
  ];
  const lastMessage = faker.lorem.paragraph();
  const lastModified = Date.now();
  const numOfNewMessages = Math.round(Math.random() * 100);
  const Wrapper = styled.div`
    width: 25rem;
  `;
  return (
    <Wrapper>
      <RoomCard
        userList={userList}
        lastMessage={lastMessage}
        lastModified={lastModified}
        numOfNewMessages={numOfNewMessages}
        />
    </Wrapper>
  );
};
