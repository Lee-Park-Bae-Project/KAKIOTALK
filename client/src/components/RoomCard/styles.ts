import styled, { css } from 'styled-components';
import Flex from 'commons/Flex';
import { color, weight } from 'styles/global';

export const grayColor = css`
  color: ${(props) => color.GRAY};
`;

export const Container = styled(Flex)`
  height: 5rem;
  margin: 0.5rem 0;
  min-width: 10rem;
  justify-content: space-between;
  &:hover{
    background: ${color.HOVER_GRAY};
  }
`;

export const RoomInfoWrapper = styled.div`
  display: flex;
  max-width: 80%;
  

`;

export const ImgWrapper = styled.div`
  margin: 1rem;
  width: auto;
`;

export const InfoWrapper = styled.div`
  flex-direction: column;
  align-items: flex-start;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  min-width: 80%;
  width: 100%;
`;

export const Row = styled.div`
  justify-content: left;
  /* min-width: 0;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  width: 80%; */
`;

export const UserListWrapper = styled.span`
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  margin-right: 1rem;
  min-width: 0;
  display: block;
  font-weight: ${weight.STRONG};
`;

export const NumWrapper = styled.span`
  ${(props) => grayColor};
  /* display: inline-flex; */

  /* color: ${color.GRAY}; */
`;

export const LastMsgWrapper = styled.div`
  ${(props) => grayColor};
  margin-right: 1rem;
  display: -webkit-box;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
  /* width: 10rem; */
  height: 2.5rem;
  overflow: hidden;
  text-overflow: ellipsis;
`;

export const SubInfoWrapper = styled(Flex)`
  flex-direction: column;
  align-items: flex-end;
  flex:1
  span{
    ${(props) => grayColor};
  }  
`;

export const Time = styled.span`
  font-size: 0.7rem;
`;
