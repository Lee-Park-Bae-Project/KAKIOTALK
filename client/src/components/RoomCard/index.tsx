import React, { FC, useState } from 'react';
import * as S from 'components/RoomCard/styles';
import Icon from 'Icon/Icon';
import Circle from 'atoms/Circle';
import moment from 'moment';
import { convertMillToMMDDYYYY } from 'common/utils';

interface Props{
  userList: string[];
  lastMessage: string;
  lastModified: number;
  numOfNewMessages: number;
  onClick?: () => void;
}
const ChatCard: FC<Props> = ({
  userList,
  lastMessage,
  lastModified,
  numOfNewMessages,
  onClick = undefined,
}) => {
  const [isSelected, setIsSelected] = useState(false);
  const numOfUser = userList.length;
  moment.locale();
  return (
    <S.Container onClick={onClick}>
      <S.RoomInfoWrapper>
        <S.ImgWrapper>
          <Icon icon="Person"/>
        </S.ImgWrapper>
        <S.InfoWrapper>
          <S.Row>
            <div>
              <S.UserListWrapper>
                {userList.join(', ')}
              </S.UserListWrapper>
            </div>


          </S.Row>

          <S.LastMsgWrapper>
            <span>{lastMessage}</span>
          </S.LastMsgWrapper>
        </S.InfoWrapper>
        <S.NumWrapper>
          {userList.length}
        </S.NumWrapper>
      </S.RoomInfoWrapper>

      <S.SubInfoWrapper>
        <S.Time>{convertMillToMMDDYYYY(lastModified)}</S.Time>
        <Circle num={numOfNewMessages}/>
      </S.SubInfoWrapper>
    </S.Container>
  );
};

export default ChatCard;
