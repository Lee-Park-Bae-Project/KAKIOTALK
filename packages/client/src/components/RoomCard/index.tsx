import React, { FC } from 'react'
import * as S from 'components/RoomCard/styles'
import Icon from 'Icon/Icon'
import {
  convertToLL, convertToLT, getCurTimeDBFormat, getDayDiff,
} from 'common/utils'

interface Props {
  participantsName: string;
  numOfParticipants: number;
  lastMessage?: string;
  lastModified?: string;
  onClick?: () => void;
}
const RoomCard: FC<Props> = ({
  participantsName,
  numOfParticipants,
  lastMessage = 'this is last message',
  lastModified = getCurTimeDBFormat(),
  onClick = undefined,
}) => {
  const dayDiffFromNow = getDayDiff(lastModified)
  return (
  <S.Container onClick={onClick}>
    <S.RoomInfoWrapper>
      <S.ImgWrapper>
        <Icon icon="Person" />
      </S.ImgWrapper>
      <S.InfoWrapper>
        <S.Row>
            <S.UserListWrapper>{participantsName}</S.UserListWrapper>
            <S.NumWrapper>{numOfParticipants}</S.NumWrapper>
        </S.Row>
        <S.LastMsgWrapper>
          {lastMessage}
        </S.LastMsgWrapper>
      </S.InfoWrapper>
    </S.RoomInfoWrapper>
    <S.SubInfoWrapper>
      <S.Time>{ dayDiffFromNow ? convertToLT(lastModified) : convertToLL(lastModified)}</S.Time>
    </S.SubInfoWrapper>
  </S.Container>
  )
}

export default RoomCard
