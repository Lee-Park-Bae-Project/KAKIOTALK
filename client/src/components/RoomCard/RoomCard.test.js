/* eslint-disable no-undef */
import React from 'react';
import { mount } from 'enzyme';
import RoomCard from 'components/RoomCard';

// faker.locale = 'ko';
describe('<RoomCard/>', () => {
  const userList = ['a',
    's',
    'd',
    'c'];
  const lastMessage = 'hihi';
  const lastModified = '2020-04-25 22:13:52';

  it('matches snapshots', () => {
    const wrapper = mount(
      <RoomCard
        userList={userList}
        lastMessage={lastMessage}
        lastModified={lastModified}
        numOfNewMessages={3}
    />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
