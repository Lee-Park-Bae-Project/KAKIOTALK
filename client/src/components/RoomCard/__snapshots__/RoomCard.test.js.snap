// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RoomCard/> matches snapshots 1`] = `
<ChatCard
  lastMessage="hihi"
  lastModified="2020-04-25 22:13:52"
  numOfNewMessages={3}
  userList={
    Array [
      "a",
      "s",
      "d",
      "c",
    ]
  }
>
  <Styled(styled.div)>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": false,
            "lastClassName": "jRfhhm",
            "rules": Array [
              "
  display: flex;
  width: 100%;
  justify-content: center;
  align-items: center;
",
              "
  height: 5rem;
  margin: 0.5rem 0;
  min-width: 10rem;
  justify-content: space-between;
  &:hover{
    background: ",
              "#F1F2F2",
              ";
  }
",
            ],
          },
          "displayName": "Styled(styled.div)",
          "foldedComponentIds": Array [
            "sc-bdVaJa",
          ],
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <div
        className="sc-bdVaJa sc-bwzfXH jRfhhm"
      >
        <styled.div>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-htpNat",
                  "isStatic": false,
                  "lastClassName": "gBEccN",
                  "rules": Array [
                    "
  display: flex;
  max-width: 80%;
",
                  ],
                },
                "displayName": "styled.div",
                "foldedComponentIds": Array [],
                "render": [Function],
                "styledComponentId": "sc-htpNat",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-htpNat gBEccN"
            >
              <styled.div>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-bxivhb",
                        "isStatic": false,
                        "lastClassName": "hznbYw",
                        "rules": Array [
                          "
  margin: 1rem;
  width: auto;
",
                        ],
                      },
                      "displayName": "styled.div",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-bxivhb",
                      "target": "div",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <div
                    className="sc-bxivhb hznbYw"
                  >
                    <Icon
                      icon="Person"
                    >
                      <ForwardRef(SvgPerson)
                        fill="currentColor"
                        height="2rem"
                        width="2rem"
                      >
                        <svg
                          fill="currentColor"
                          height="2rem"
                          width="2rem"
                        >
                          person.svg
                        </svg>
                      </ForwardRef(SvgPerson)>
                    </Icon>
                  </div>
                </StyledComponent>
              </styled.div>
              <styled.div>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-ifAKCX",
                        "isStatic": false,
                        "lastClassName": "ebQtyy",
                        "rules": Array [
                          "
  flex-direction: column;
  align-items: flex-start;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  min-width: 80%;
  width: 100%;
",
                        ],
                      },
                      "displayName": "styled.div",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-ifAKCX",
                      "target": "div",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <div
                    className="sc-ifAKCX ebQtyy"
                  >
                    <styled.div>
                      <StyledComponent
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "sc-EHOje",
                              "isStatic": false,
                              "lastClassName": "elLoSm",
                              "rules": Array [
                                "
  justify-content: left;
",
                              ],
                            },
                            "displayName": "styled.div",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "sc-EHOje",
                            "target": "div",
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                      >
                        <div
                          className="sc-EHOje elLoSm"
                        >
                          <div>
                            <styled.span>
                              <StyledComponent
                                forwardedComponent={
                                  Object {
                                    "$$typeof": Symbol(react.forward_ref),
                                    "attrs": Array [],
                                    "componentStyle": ComponentStyle {
                                      "componentId": "sc-bZQynM",
                                      "isStatic": false,
                                      "lastClassName": "ierfGz",
                                      "rules": Array [
                                        "
  text-overflow: ellipsis;
  overflow: hidden;
  white-space: nowrap;
  margin-right: 1rem;
  min-width: 0;
  display: block;
  font-weight: ",
                                        "800",
                                        ";
",
                                      ],
                                    },
                                    "displayName": "styled.span",
                                    "foldedComponentIds": Array [],
                                    "render": [Function],
                                    "styledComponentId": "sc-bZQynM",
                                    "target": "span",
                                    "toString": [Function],
                                    "warnTooManyClasses": [Function],
                                    "withComponent": [Function],
                                  }
                                }
                                forwardedRef={null}
                              >
                                <span
                                  className="sc-bZQynM ierfGz"
                                >
                                  a, s, d, c
                                </span>
                              </StyledComponent>
                            </styled.span>
                          </div>
                        </div>
                      </StyledComponent>
                    </styled.div>
                    <styled.div>
                      <StyledComponent
                        forwardedComponent={
                          Object {
                            "$$typeof": Symbol(react.forward_ref),
                            "attrs": Array [],
                            "componentStyle": ComponentStyle {
                              "componentId": "sc-htoDjs",
                              "isStatic": false,
                              "lastClassName": "jNcOQg",
                              "rules": Array [
                                "
  ",
                                [Function],
                                ";
  margin-right: 1rem;
  display: -webkit-box;
  -webkit-line-clamp: 2;
  -webkit-box-orient: vertical;
  height: 2.5rem;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: normal;
",
                              ],
                            },
                            "displayName": "styled.div",
                            "foldedComponentIds": Array [],
                            "render": [Function],
                            "styledComponentId": "sc-htoDjs",
                            "target": "div",
                            "toString": [Function],
                            "warnTooManyClasses": [Function],
                            "withComponent": [Function],
                          }
                        }
                        forwardedRef={null}
                      >
                        <div
                          className="sc-htoDjs jNcOQg"
                        >
                          <span>
                            hihi
                          </span>
                        </div>
                      </StyledComponent>
                    </styled.div>
                  </div>
                </StyledComponent>
              </styled.div>
              <styled.span>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-gzVnrw",
                        "isStatic": false,
                        "lastClassName": "gJorOU",
                        "rules": Array [
                          "
  ",
                          [Function],
                          ";
",
                        ],
                      },
                      "displayName": "styled.span",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-gzVnrw",
                      "target": "span",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <span
                    className="sc-gzVnrw gJorOU"
                  >
                    4
                  </span>
                </StyledComponent>
              </styled.span>
            </div>
          </StyledComponent>
        </styled.div>
        <Styled(styled.div)>
          <StyledComponent
            forwardedComponent={
              Object {
                "$$typeof": Symbol(react.forward_ref),
                "attrs": Array [],
                "componentStyle": ComponentStyle {
                  "componentId": "sc-dnqmqq",
                  "isStatic": false,
                  "lastClassName": "exgQnf",
                  "rules": Array [
                    "
  display: flex;
  width: 100%;
  justify-content: center;
  align-items: center;
",
                    "
  flex-direction: column;
  align-items: flex-end;
  flex:1
  span {
    ",
                    [Function],
                    ";
  }  
",
                  ],
                },
                "displayName": "Styled(styled.div)",
                "foldedComponentIds": Array [
                  "sc-bdVaJa",
                ],
                "render": [Function],
                "styledComponentId": "sc-dnqmqq",
                "target": "div",
                "toString": [Function],
                "warnTooManyClasses": [Function],
                "withComponent": [Function],
              }
            }
            forwardedRef={null}
          >
            <div
              className="sc-bdVaJa sc-dnqmqq exgQnf"
            >
              <styled.span>
                <StyledComponent
                  forwardedComponent={
                    Object {
                      "$$typeof": Symbol(react.forward_ref),
                      "attrs": Array [],
                      "componentStyle": ComponentStyle {
                        "componentId": "sc-iwsKbI",
                        "isStatic": false,
                        "lastClassName": "gnPtlj",
                        "rules": Array [
                          "
  font-size: 0.7rem;
",
                        ],
                      },
                      "displayName": "styled.span",
                      "foldedComponentIds": Array [],
                      "render": [Function],
                      "styledComponentId": "sc-iwsKbI",
                      "target": "span",
                      "toString": [Function],
                      "warnTooManyClasses": [Function],
                      "withComponent": [Function],
                    }
                  }
                  forwardedRef={null}
                >
                  <span
                    className="sc-iwsKbI gnPtlj"
                  >
                    04-25-2020
                  </span>
                </StyledComponent>
              </styled.span>
              <Circle
                num={3}
              >
                <styled.div>
                  <StyledComponent
                    forwardedComponent={
                      Object {
                        "$$typeof": Symbol(react.forward_ref),
                        "attrs": Array [],
                        "componentStyle": ComponentStyle {
                          "componentId": "sc-gZMcBi",
                          "isStatic": false,
                          "lastClassName": "iFUJoV",
                          "rules": Array [
                            "
    display: flex;
    align-items: center;
    justify-content: center;
    width: fit-content;
    height: 1rem;
    margin: 1rem;
    padding: 0.1rem 0.5rem;
    border-radius: 2rem;
    text-align: center;
    font-size: ",
                            "0.8rem",
                            ";
    background-color: ",
                            "#DD6F5A",
                            ";
    color: ",
                            "#FEFFFF",
                            ";
    font-weight: ",
                            "600",
                            ";
  ",
                          ],
                        },
                        "displayName": "styled.div",
                        "foldedComponentIds": Array [],
                        "render": [Function],
                        "styledComponentId": "sc-gZMcBi",
                        "target": "div",
                        "toString": [Function],
                        "warnTooManyClasses": [Function],
                        "withComponent": [Function],
                      }
                    }
                    forwardedRef={null}
                  >
                    <div
                      className="sc-gZMcBi iFUJoV"
                    >
                      3
                    </div>
                  </StyledComponent>
                </styled.div>
              </Circle>
            </div>
          </StyledComponent>
        </Styled(styled.div)>
      </div>
    </StyledComponent>
  </Styled(styled.div)>
</ChatCard>
`;
